libs/sdk-ui-ext/src/internal/interfaces/VisualizationDescriptor.ts

```diff
@@ -0,0 +1,56 @@
+// (C) 2021 GoodData Corporation
+import { IInsightDefinition } from "@gooddata/sdk-model";
+import { IVisConstruct, IVisualization } from "./Visualization";
+
+/**
+ * Factories that create a new instance of pluggable visualization.
+ *
+ * @alpha
+ */
+export type PluggableVisualizationFactory = (params: IVisConstruct) => IVisualization;
+
+interface ISizeInfo {
+    default?: number;
+    min?: number;
+    max?: number;
+    defaultPx?: number;
+    minPx?: number;
+    maxPx?: number;
+}
+
+export interface IVisualizationSizeInfo {
+    width: ISizeInfo;
+    height: ISizeInfo;
+}
+
+export interface ILayoutDescriptor {
+    gridColumnsCount: number;
+    gridRowHeight: number;
+    layoutContainerSize?: {
+        width: number;
+        height: number;
+    };
+    /**
+     * Converts height in px to the height in layout units
+     */
+    toGridHeight(heightPx: number): number;
+}
+
+/**
+ * @alpha
+ */
+export interface IVisualizationDescriptor {
+    /**
+     * Gets pluggable visualization factory from visualization descriptor.
+     *
+     * @alpha
+     */
+    getFactory(): PluggableVisualizationFactory;
+
+    // TODO INE should size info be in grid units or px?
+    getSizeInfo(
+        insight: IInsightDefinition,
+        layoutDescriptor: ILayoutDescriptor,
+        enableCustomHeight: boolean,
+    ): IVisualizationSizeInfo;
+}
```

Comment:
we need contextual and contractual type of information here:
- Who uses these descriptors and when and for what purpose
- The various return types, we need to document how does the information impact the layouting. What will the layouter do if value is such-and-such
- For example ISizeInfo. all parameters are optional. the type needs to document their purpose and what happens if the value is not specified etc.

Look at from perspective of a person who gets these interfaces and needs to implement them for their pluggable visualization. The interface should have enough documentation for them to understand what's going on.

(added benefit is that when we go to this code we can learn how KD _should_ be processing the information)